src/tests/ascii/test_ascii.f90

```diff
@@ -540,4 +545,72 @@ pure logical function validation_func_interface(c)
         write(*,'(5X,12(I4))') (count(table(:,i)),i=1,12)
     end subroutine test_ascii_table
 
+    subroutine test_to_lower_string
+        character(len=:), allocatable :: dlc
+        character(len=32), parameter :: input = "UPPERCASE"
+
+        dlc = to_lower("UPPERCASE")
+        call check(dlc == "uppercase")
+
+        dlc = to_lower(input)
+        call check(len(dlc) == 32)
+        call check(len_trim(dlc) == 9)
+        call check(trim(dlc) == "uppercase")
+
+        dlc = to_lower("0123456789ABCDE")
+        call check(dlc == "0123456789abcde")
+    end subroutine test_to_lower_string
+
+    subroutine test_to_upper_string
+        character(len=:), allocatable :: dlc
+        character(len=32), parameter :: input = "lowercase"
+
+        dlc = to_upper("lowercase")
+        call check(dlc == "LOWERCASE")
+
+        dlc = to_upper(input)
+        call check(len(dlc) == 32)
+        call check(len_trim(dlc) == 9)
+        call check(trim(dlc) == "LOWERCASE")
+
+        dlc = to_upper("0123456789abcde")
+        call check(dlc == "0123456789ABCDE")
+    end subroutine test_to_upper_string
+
+    subroutine test_to_title_string
+        character(len=:), allocatable :: dlc
+        character(len=32), parameter :: input = "tiTLe"
+
+        dlc = to_title("tiTLe")
+        call check(dlc == "Title")
+
+        dlc = to_title(input)
+        call check(len(dlc) == 32)
+        call check(len_trim(dlc) == 5)
+        call check(trim(dlc) == "Title")
+
+        dlc = to_title(" s P a C e D !")
+        call check(dlc == " S p a c e d !")
+
+        dlc = to_title("1st, 2nd, 3rd")
+        call check(dlc == "1st, 2nd, 3rd")
+
+        dlc = to_title("""quOTed""")
+        call check(dlc == """Quoted""")
+    end subroutine test_to_title_string
+
+    subroutine test_reverse_string
+        character(len=:), allocatable :: dlc
+        character(len=32), parameter :: input = "reversed"
+
+        dlc = reverse("reversed")
+        call check(dlc == "desrever")
+
+        dlc = reverse(input)
+        call check(len(dlc) == 32)
+        call check(len_trim(dlc) == 32)
+        call check(trim(dlc) == "                        desrever")
+        call check(trim(adjustl(dlc)) == "desrever")
```

Comment:
If I am correct, in Fortran 90, two strings are defined to be equal if they have the
exact same contents, except for trailing blanks, i.e.,
```fortran
"abc" == "abc    " !.true.
```
Therefore, I would expect that `reverse('abc') == reverse('abc    ')`.

The current behaviour is ok for me, if it would be mentioned in the specs.

Reply:
Trailing whitespace is somewhat special, while leading whitespace is not. With reverse I would expect

```f90
call check(dlc == reverse(reverse(dlc)))
```

to always hold. Following the proposal to handle trailing whitespace special in reverse we would arrive at:

```f90
character(len=*), parameter :: input = "  abc "
character(len=:), allocatable :: dlc
dlc = reverse(input)
! dlc == "cba   "
dlc = reverse(dlc)
! dlc == "abc   " .and. .not. (dlc == input)
```

Reply:
In a previous comment I also alluded to the second interpretation of reverse which left aligns.

I agree with @awvwgk the specification such that
```
call check(dlc == reverse(reverse(dlc)))
```
is less surprising. It is also in agreement with at least one other language (Python).